python/lsst/ts/observatory/control/auxtel/atcalsys.py

```diff
@@ -0,0 +1,421 @@
+        logging in the class use the script logging.
+    intended_usage: `int`
+        Optional integer that maps to a list of intended operations. This is
+        used to limit the resources allocated by the class by gathering some
+        knowledge about the usage intention. By default allocates all
+        resources.
+    """
+
+    def __init__(
+        self,
+        domain: typing.Optional[salobj.Domain] = None,
+        log: typing.Optional[logging.Logger] = None,
+        intended_usage: typing.Optional[int] = None,
+    ) -> None:
+
+        self.electrometer_index = 1
+        self.fiberspectrograph_index = 1
+
+        super().__init__(
+            components=[
+                "ATCamera",  # isn't this just LATISS?
+                "ATMonochromator",
+                "ATWhiteLight",
+                f"FiberSpectrograph:{self.fiberspectrograph_index}",
+                f"Electrometer:{self.electrometer_index}",
+            ],
+            instrument_setup_attributes=["filter", "calib_type"],
+            domain=domain,
+            log=log,
+            intended_usage=intended_usage,
+        )
+
+        self.entrance_slit = 0.5
+        self.exit_slit = 0.5
+
+        # White lamp config
+        self.timeout_lamp_warm_up = 60 * 15
+        self.track_lamp_warmup = 60
+        self.cmd_timeout = 30
+        self.timeout_std = 20
+        self.whitelight_startup_power = 1000
+
+        # Chiller config
+        self.timeout_chiller_cool_down = 60 * 15
+        self.chiller_temp_tolerance_relative = 0.2
+        self.chiller_temperature = 20
+
+        self.filter_gratings = {
+            "u": 0,  # "Blue"
+            "g": 0,
+            "r": 0,
+            "i": 1,  # "Red"
+            "z": 1,
+            "y": 1,
+        }
+        self.exptime_dict = (
+            {  # I don't know how to instantiate this so that I get dict[str: float]
+                "camera": 0.0,
+                "electrometer": 0.0,
+                "fiberspectrograph": 0.0,
+            }
+        )
+        self.delay = 1
+
+        self.ready_for_flats = False
+
+    async def warmup_lightsource(self, calib_type: str) -> None:
+        """Turn on the Chiller and the ATWhiteLight and wait for the
+        warmup to complete
+
+        Parameters
+        ----------
+        calib_type : `str`
+            Identifies if White Light or Monochromatic. For the ATCalsys,
+            the light sourceis the same. However, if it is White Light
+            configure the Monochromator now.
+
+        Raises
+        ------
+        RuntimeError:
+
+        """
+        self.calib_type = calib_type
+        try:
+            await self.start_chiller()
+            await self.wait_for_chiller_temp_within_tolerance()
+            await self.switch_lamp_on()
+            await self.wait_for_lamp_to_warm_up()
+            self.ready_for_flats = True
+        except asyncio.TimeoutError:
```

Comment:
I think in this case it might be better to not wrap the calls in try/except. By doing this you are hiding the actual error and it will be harder to debug later.

Reply:
Ok. all try/excepts have been removed